### PR TITLE
Limit PHPUnit version / PHPUnit 6 problems

### PR DESCRIPTION
**Pull request by SpacePossum**
Trying to figure out a nice way to support PHPUnit 6 without BC breaks.
Doesn't seem possible because of the changes done in the Contraints, so setting the supporting PHPUnit version limit seems needed.

Making PHPUnit as package seems logical to me (https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2757), like to hear from others though (ping @keradus ;) )

### CODE REVIEW

**Comment by SpacePossum** [comment]
TODO
* PHPUnit_Framework_Exception
* PHPUnit_Framework_Constraint
* PHPUnit_Framework_Constraint_Not
* PHPUnit_Framework_Constraint_IsType
* PHPUnit_Framework_Constraint_IsIdentical

File: tests/PHPUnit/Tests/GeckoTestCase.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ * (c) GeckoPackages https://github.com/GeckoPackages
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+class_exists('\PHPUnit\Framework\TestCase')
+    ? eval('abstract class GeckoTestCase extends \PHPUnit\Framework\TestCase{}')
+    : eval('abstract class GeckoTestCase extends \PHPUnit_Framework_TestCase{}')
+;
```
Comment (keradus) [inline tests/PHPUnit/Tests/GeckoTestCase.php]: aaaaa..... no

**Comment by keradus** [comment]
So, I think you need to make clear statement what is the purpose of this packages.
Adding new features or backporting PHPUnit native features to older version. I strongly suggest to not do both in one package.

Example: there is `assertFileExists` assertion, which is already PHPUnit native assertion.
If reason is to provide new features, maybe you could send them to PHPUnit directly ?
If backporting, it's needed to be analysed what is worth to be backported, as I think it's worthless to backport 5/6 features into 4.x

File: tests/PHPUnit/Tests/GeckoTestCase.php [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+/*
+ * This file is part of the GeckoPackages.
+ *
+ * (c) GeckoPackages https://github.com/GeckoPackages
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+class_exists('\PHPUnit\Framework\TestCase')
```
Comment (GrahamCampbell) [inline tests/PHPUnit/Tests/GeckoTestCase.php]: Leading slash should not be used. PHP just deletes it straight away.

**Comment by SpacePossum** [comment]
older PHPUnit failure on travis fixed @ https://github.com/GeckoPackages/GeckoPHPUnit/pull/34
PHPUnit 6 support will be done in other PR

**Pull request closed without merging**